libsolidity/ast/Types.cpp

```diff
@@ -2206,7 +2206,10 @@ FunctionType::FunctionType(VariableDeclaration const& _varDecl):
 				if (auto arrayType = dynamic_cast<ArrayType const*>(member.type.get()))
 					if (!arrayType->isByteArray())
 						continue;
-				m_returnParameterTypes.push_back(member.type);
+				m_returnParameterTypes.push_back(ReferenceType::copyForLocationIfReference(
```

Comment:
Hm, can we have a test for this?

Reply:
It only turned up when I activated the strict decoder, not sure this is testable.

Reply:
Ah ok, right, I'll add a more direct test.